src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderRangedAI.java

```diff
@@ -163,7 +163,7 @@ protected PathResult moveInAttackPosition(final LivingEntity target)
     @Override
     protected boolean isAttackableTarget(final LivingEntity target)
     {
-        return target instanceof EntityCitizen || (target instanceof Player && !((Player) target).isCreative() && !target.isSpectator());
+        return (target instanceof EntityCitizen && !target.isInvisible()) || (target instanceof Player && !((Player) target).isCreative() && !target.isSpectator());
```

Comment:
Another option might be to use `TargetingConditions` here, which can do partial invisibility based on range if we want to distinguish between invisibility potions and away-in-nether.